WordPress/src/main/java/org/wordpress/android/ui/reader/repository/usecases/PostLikeActionUseCase.kt

```diff
@@ -0,0 +1,54 @@
+package org.wordpress.android.ui.reader.repository.usecases
+
+import kotlinx.coroutines.CoroutineDispatcher
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode.BACKGROUND
+import org.wordpress.android.models.ReaderPost
+import org.wordpress.android.modules.IO_THREAD
+import org.wordpress.android.ui.reader.actions.ReaderPostActions
+import org.wordpress.android.ui.reader.repository.OnReaderRepositoryEvent
+import org.wordpress.android.ui.reader.repository.OnReaderRepositoryEvent.OnPostLikeEnded
+import org.wordpress.android.util.EventBusWrapper
+import javax.inject.Inject
+import javax.inject.Named
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class PostLikeActionUseCase @Inject constructor(
+    @Named(IO_THREAD) private val ioDispatcher: CoroutineDispatcher,
+    private val eventBusWrapper: EventBusWrapper
+) : ReaderRepositoryDispatchingUseCase(ioDispatcher) {
+    private var continuation: kotlin.coroutines.Continuation<OnReaderRepositoryEvent>? = null
```

Comment:
We might want to consider keeping a list/array/map of continuations so we can support multiple parallel calls, wdyt?

Reply:
That's a great idea ... was only thinking single with this draft, but that makes a lot of sense.

Reply:
In commit aec3678